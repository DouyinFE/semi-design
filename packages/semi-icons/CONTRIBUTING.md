## ðŸ“£ Introduction

How to contribute icon to this package

##  ðŸš€ Steps
- Add the svg file under the path ```src/svgs/```, naming references to other files.
- Run command ```lerna run build:icon --scope @douyinfe/semi-icons```, the corresponding tsx file will be generated under the path ```src/icons```.
- Add the category and name of the new icon to the ```src/svgs/meta.json``` file.
- Finally, run command ```yarn docsite``` and go to the icon page to confirm whether the icon is successfully added.