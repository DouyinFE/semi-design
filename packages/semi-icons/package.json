{
  "name": "@douyinfe/semi-icons",
  "version": "2.16.1",
  "description": "semi icons",
  "keywords": [
    "semi",
    "icons"
  ],
  "author": "",
  "homepage": "",
  "main": "lib/cjs/index.js",
  "module": "lib/es/index.js",
  "typings": "lib/es/index.d.ts",
  "license": "MIT",
  "files": [
    "lib",
    "src",
    "dist"
  ],
  "sideEffects": [
    "*.scss",
    "*.css",
    "lib/es/index.js"
  ],
  "scripts": {
    "clean": "rimraf dist lib",
    "build:icon": "node scripts/build-icon ",
    "build:lib": "node ./scripts/compileLib.js",
    "build:js": "npm run build:lib && node scripts/compileDist.js",
    "prepublishOnly": "npm run clean && npm run build:js"
  },
  "dependencies": {
    "@babel/runtime-corejs3": "^7.15.4",
    "classnames": "^2.2.6"
  },
  "devDependencies": {
    "@babel/plugin-proposal-decorators": "^7.15.8",
    "@babel/plugin-transform-runtime": "^7.15.8",
    "@babel/preset-env": "^7.15.8",
    "@babel/preset-react": "^7.14.5",
    "@douyinfe/semi-webpack-plugin": "2.16.1",
    "babel-loader": "^8.2.2",
    "css-loader": "4.3.0",
    "del": "^6.0.0",
    "gulp": "^4.0.2",
    "gulp-babel": "^8.0.0",
    "gulp-replace": "^1.1.3",
    "gulp-sass": "^5.0.0",
    "gulp-typescript": "^6.0.0-alpha.1",
    "merge2": "^1.4.1",
    "mini-css-extract-plugin": "0.11.3",
    "rimraf": "^3.0.2",
    "sass": "1.32.2",
    "sass-loader": "^10.0.5",
    "terser-webpack-plugin": "^4.2.3",
    "through2": "^4.0.2",
    "ts-loader": "^5.4.5",
    "webpack": "^4.46.0"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0 || ^18.0.0"
  },
  "_unpkg": true,
  "unpkgFiles": [
    "*"
  ],
  "gitHead": "eb34a4f25f002bb4cbcfa51f3df93bed868c831a"
}
