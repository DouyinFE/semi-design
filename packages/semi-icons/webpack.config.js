const path = require('path');
const webpack = require('webpack');
const TerserPlugin = require('terser-webpack-plugin');
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const CaseSensitivePathsPlugin = require('case-sensitive-paths-webpack-plugin');

module.exports = function getWebpackConfig({ minimize }){
    return {
        mode: 'production',
        bail: true,
        devtool: 'source-map',
        entry: {
            index: ['./lib/es/index.js']
        },
        output: {
            filename: minimize ? 'umd/semi-icons.min.js' : 'umd/semi-icons.js',
            path: path.join(__dirname, 'dist'),
            library: 'SemiIcons',
            libraryTarget: 'umd'
        },
        module: {
            rules: [
                { 
                    test: /\.css$/, 
                    loaders: [
                        {
                            loader: MiniCssExtractPlugin.loader
                        },
                        {
                            loader: 'css-loader'
                        }
                    ] 
                },
            ]
        },
        optimization: {
            minimize: !!minimize,
            minimizer: [new TerserPlugin()]
        },
        performance: { maxEntrypointSize: 10485760, maxAssetSize: 5242880 },
        plugins: [
            new webpack.DefinePlugin({
                'process.env': { NODE_ENV: '"production"' }
            }),
            new CaseSensitivePathsPlugin(),
            new webpack.HashedModuleIdsPlugin(),
            new MiniCssExtractPlugin({
                // Options similar to the same options in webpackOptions.output
                // both options are optional
                filename: minimize ? 'css/semi-icons.min.css' : 'css/semi-icons.css',
            })
        ],
        resolve: {
            extensions: ['.ts', '.tsx', '.js', '.jsx', '.json']
        },
        externals: {
            react: {
                root: 'React',
                commonjs2: 'react',
                commonjs: 'react',
                amd: 'react'
            }
        }
    };
};