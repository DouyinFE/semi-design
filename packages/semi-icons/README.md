##  📣 Introduce

Semantic Icons design by Semi Team

## 🏷 Features

- 330+ Icons
- Filled and Stroked Icons
- Support custom color

##  🚀 Getting Start

### install deps

```shell
npm install @douyinfe/semi-icons
```

### as a react component

```tsx
import { IconSend } from '@douyinfe/semi-icons';

ReactDOM.render(<IconSend />, root);
```

