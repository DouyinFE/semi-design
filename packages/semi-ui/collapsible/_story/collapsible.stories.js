/* argus-disable unPkgSensitiveInfo */
import React, { useState } from 'react';
import { Button, CheckboxGroup, Upload, Table, Collapse, Tabs } from '../../index';
import Collapsible from '..';
import NestedDemo from './Nested';
import { IconChevronDown, IconChevronRight, IconUpload } from '@douyinfe/semi-icons';

const TabPane = Tabs.TabPane;

export default {
  title: 'Collapsible',
}

class Demo extends React.Component {
  state = {
    isOpen: false,
  };
  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  };

  render() {
    const { isOpen } = this.state;
    const collapsed = (
      <ul>
        <li>
          <p>Semi Design 以内容优先进行设计。</p>
        </li>
        <li>
          <p>更容易地自定义主题。</p>
        </li>
        <li>
          <p>适用国际化场景。</p>
        </li>
        <li>
          <p>效率场景加入人性化关怀。</p>
        </li>
      </ul>
    );
    return (
      <div>
        <Button onClick={() => this.toggle()}>显示更多</Button>
        <Collapsible isOpen={isOpen}>{collapsed}</Collapsible>
      </div>
    );
  }
}

export const RegularCollapsible = () => <Demo />;

class DemoDOM extends React.Component {
  state = {
    isOpen: false,
  };
  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  };

  render() {
    const { isOpen } = this.state;
    const collapsed = (
      <ul>
        <li>
          <p>Semi Design 以内容优先进行设计。</p>
        </li>
        <li>
          <p>更容易地自定义主题。</p>
        </li>
        <li>
          <p>适用国际化场景。</p>
        </li>
        <li>
          <p>效率场景加入人性化关怀。</p>
        </li>
      </ul>
    );
    return (
      <div>
        <Button onClick={() => this.toggle()}>显示更多</Button>
        <Collapsible keepDOM isOpen={isOpen}>
          {collapsed}
        </Collapsible>
      </div>
    );
  }
}

export const KeepDom = () => <DemoDOM />;

class DefaultOpen extends React.Component {
  state = {
    isOpen: true,
  };
  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  };

  render() {
    const { isOpen } = this.state;
    const collapsed = (
      <ul>
        <li>
          <p>Semi Design 以内容优先进行设计。</p>
        </li>
        <li>
          <p>更容易地自定义主题。</p>
        </li>
        <li>
          <p>适用国际化场景。</p>
        </li>
        <li>
          <p>效率场景加入人性化关怀。</p>
        </li>
      </ul>
    );
    return (
      <div>
        <Button onClick={() => this.toggle()}>toggle</Button>
        <Collapsible isOpen={isOpen}>{collapsed}</Collapsible>
      </div>
    );
  }
}

export const DefaultOpenDemo = () => <DefaultOpen />;
export const NestedCollapsible = () => <NestedDemo />;

class Wrap extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: props.isOpen,
      options: [
        {
          label: '抖音',
          value: 'dy',
        },
        {
          label: '火山',
          value: 'hotsoon',
        },
        {
          label: '皮皮虾',
          value: 'pipixia',
        },
        {
          label: '今日头条',
          value: 'toutiao',
        },
      ],
      values: [],
    };
    this.toggle = this.toggle.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  toggle() {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  }

  onChange(values) {
    this.setState({
      values,
    });
  }

  render() {
    const { isOpen, options, values } = this.state;
    return (
      <>
        <div onClick={this.toggle}>
          {isOpen ? <IconChevronDown /> : <IconChevronRight />}
          权限点
          <span>
            {values.length}/{options.length}
          </span>
        </div>
        <Collapse defaultActiveKey="1" motion={false}>
          <Collapse.Panel header="This is panel header 1" itemKey="1">
            <p>Hi, bytedance dance dance. This is the docsite of Semi UI. </p>
          </Collapse.Panel>
          <Collapse.Panel header="This is panel header 2" itemKey="2">
            <p>Hi, bytedance dance dance. This is the docsite of Semi UI. </p>
          </Collapse.Panel>
          <Collapse.Panel header="This is panel header 3" itemKey="3">
            <p>Hi, bytedance dance dance. This is the docsite of Semi UI. </p>
          </Collapse.Panel>
        </Collapse>
        {/* <Collapsible isOpen={isOpen}>
           <div style={{ height: '40px' }}>
               <CheckboxGroup options={options} direction="horizontal" onChange={this.onChange} value={values} />
           </div>
        </Collapsible> */}
      </>
    );
  }
}

export const CollapsibleTest = () => <Wrap />;

class App extends React.Component {
  render() {
    const expandColumns = [
      {
        title: 'Name',
        dataIndex: 'name',
        key: 'name',
      },
      {
        title: 'Age',
        dataIndex: 'age',
        key: 'age',
      },
      {
        title: 'Address',
        dataIndex: 'address',
        key: 'address',
      },
      {
        title: 'Action',
        dataIndex: '',
        key: 'x',
        render: () => <a>Delete</a>,
      },
    ];
    const rowSelection = {
      onChange: (selectedRowKeys, selectedRows) => {
        // console.log(`selectedRowKeys: ${selectedRowKeys}`, 'selectedRows: ', selectedRows);
      },
      getCheckboxProps: record => ({
        disabled: record.name === 'Michael James',
        // Column configuration not to be checked
        name: record.name,
      }),
    };
    const expandData = [
      {
        name: 'John Brown',
        age: 32,
        address: 'New York No. 1 Lake Park',
        description:
          'My name is John Brown, I am 32 years old, living in New York No. 1 Lake Park.',
      },
      {
        name: 'Jim Green',
        age: 42,
        address: 'London No. 1 Lake Park',
        description: 'My name is Jim Green, I am 42 years old, living in London No. 1 Lake Park.',
      },
      {
        name: 'Joe Black',
        age: 32,
        address: (
          <Collapse defaultActiveKey="1" motion={false}>
            <Collapse.Panel header="This is panel header 1" itemKey="1">
              <p>Hi, bytedance dance dance. This is the docsite of Semi UI. </p>
            </Collapse.Panel>
            <Collapse.Panel header="This is panel header 2" itemKey="2">
              <p>Hi, bytedance dance dance. This is the docsite of Semi UI. </p>
            </Collapse.Panel>
            <Collapse.Panel header="This is panel header 3" itemKey="3">
              <p>Hi, bytedance dance dance. This is the docsite of Semi UI. </p>
            </Collapse.Panel>
          </Collapse>
        ),
        description: 'My name is Joe Black, I am 32 years old, living in Sidney No. 1 Lake Park.',
      },
    ];

    const expandRowRender = (record, index) => <Wrap isOpen={true} />;

    return (
      <Table
        rowKey={'name'}
        columns={expandColumns}
        rowSelection={rowSelection}
        expandedRowRender={expandRowRender}
        dataSource={expandData}
      />
    );
  }
}

export const CollapsibleInTable = () => <App />;

class InTab extends React.Component {
  constructor() {
    super();
    this.state = {
      isOpenFirst: false,
      isOpenSecond: false,
    };
  }

  render() {
    const { isOpenFirst, isOpenSecond } = this.state;
    return (
      <div>
        <Tabs
          onTabClick={e =>
            this.setState({
              active: e,
            })
          }
        >
          <TabPane tab="第一" itemKey="1">
            <h3>第一个tabpane</h3>
            <Button
              onClick={() => {
                this.setState({
                  isOpenFirst: !isOpenFirst,
                });
              }}
            >
              开关
            </Button>
            <Collapsible isOpen={isOpenFirst}>
              第一个tabpane下的collapsible的open状态正常
            </Collapsible>
          </TabPane>
          <TabPane tab="第二" itemKey="2">
            <h3>第二个tabpane</h3>
            <Button
              onClick={() => {
                this.setState({
                  isOpenSecond: !isOpenSecond,
                });
              }}
            >
              开关
            </Button>
            <Collapsible isOpen={isOpenSecond}>
              第二个tabpane下的collapsibleopen状态异常
            </Collapsible>
          </TabPane>
        </Tabs>
      </div>
    );
  }
}

export const CollapsibleInTab = () => <InTab />;

CollapsibleInTab.story = {
  name: 'collapsible in tab',
};

class WithUpload extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: true,
    };
    this.toggle = this.toggle.bind(this);
    this.action = '//semi.design/api/upload/';
    this.defaultFileList = [
      {
        preview: false,
        name: '2D (2).ecpj',
        status: 'success',
        uid: 'd116a179410eb0ca18e66074509bde93-0',
        url:
          'https://sf6-cdn-tos.douyinstatic.com/obj/eden-cn/ptlz_zlp/ljhwZthlaukjlkulzlp/root-web-sites/bf8647bffab13c38772c9ff94bf91a9d.jpg',
      },
      {
        preview: false,
        name: '2D-sticker-temp.psd',
        status: 'success',
        uid: 'b7d579069320590ba4b128672eedbae2-1',
        url:
          'https://sf6-cdn-tos.douyinstatic.com/obj/eden-cn/ptlz_zlp/ljhwZthlaukjlkulzlp/root-web-sites/bf8647bffab13c38772c9ff94bf91a9d.jpg',
      },
      {
        preview: false,
        name: '2D-sticker-temp (1).psd',
        status: 'success',
        uid: 'b7d579069320590ba4b128672eedbae2-2',
        url:
          'https://sf6-cdn-tos.douyinstatic.com/obj/eden-cn/ptlz_zlp/ljhwZthlaukjlkulzlp/root-web-sites/bf8647bffab13c38772c9ff94bf91a9d.jpg',
      },
    ];
  }

  toggle() {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  }

  render() {
    const { isOpen } = this.state;
    return (
      <div>
        <Collapsible isOpen={isOpen} collapseHeight={120}>
          <Upload
            dragable={true}
            name="file" // accept={ALLOW_FILE}
            defaultFileList={[
              {
                preview: false,
                name: '2D (2).ecpj',
                status: 'success',
                uid: 'd116a179410eb0ca18e66074509bde93-0',
                url:
                  'https://sf6-cdn-tos.douyinstatic.com/obj/eden-cn/ptlz_zlp/ljhwZthlaukjlkulzlp/root-web-sites/bf8647bffab13c38772c9ff94bf91a9d.jpg',
              },
              {
                preview: false,
                name: '2D-sticker-temp.psd',
                status: 'success',
                uid: 'b7d579069320590ba4b128672eedbae2-1',
                url:
                  'https://sf6-cdn-tos.douyinstatic.com/obj/eden-cn/ptlz_zlp/ljhwZthlaukjlkulzlp/root-web-sites/bf8647bffab13c38772c9ff94bf91a9d.jpg',
              },
              {
                preview: false,
                name: '2D-sticker-temp (1).psd',
                status: 'success',
                uid: 'b7d579069320590ba4b128672eedbae2-2',
                url:
                  'https://sf6-cdn-tos.douyinstatic.com/obj/eden-cn/ptlz_zlp/ljhwZthlaukjlkulzlp/root-web-sites/bf8647bffab13c38772c9ff94bf91a9d.jpg',
              },
            ]}
            dragMainText="点击上传文件或拖拽文件到这里"
            dragSubText="图片、PDF、PPT、Word、视频等"
            onError={this.uploadError}
            onSuccess={(...args) => this.uploadSuccess(i, ...args)}
            onRemove={(...args) => this.onRemoveFile(i, ...args)}
            onProgress={() =>
              this.setState({
                fileLoading: true,
              })
            }
          >
            <Button icon={<IconUpload />} theme="light">
              点击上传
            </Button>
          </Upload>
        </Collapsible>
        <Button onClick={this.toggle}>{isOpen ? '剩余1项' : '展开'}</Button>
      </div>
    );
  }
}

export const CollapsibleWithUpload = () => <WithUpload />;

class CusHeight extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
    };
    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  }

  render() {
    const { isOpen } = this.state;
    const maskStyle = isOpen
      ? {}
      : {
          WebkitMaskImage:
            'linear-gradient(to bottom, black 0%, rgba(0, 0, 0, 1) 60%, rgba(0, 0, 0, 0.2) 80%, transparent 100%)',
        };
    const collapsed = (
      <ul>
        <li>
          <p>Semi Design 以内容优先进行设计。</p>
        </li>
        <li>
          <p>更容易地自定义主题。</p>
        </li>
        <li>
          <p>适用国际化场景。</p>
        </li>
        <li>
          <p>效率场景加入人性化关怀。</p>
        </li>
      </ul>
    );
    const linkStyle = {
      position: 'absolute',
      left: 0,
      right: 0,
      textAlign: 'center',
      bottom: -10,
      fontWeight: 700,
      cursor: 'pointer',
    };
    console.log('out state', isOpen);
    return (
      <>
        <Button onClick={this.toggle}>Toggle</Button>
        <div
          style={{
            position: 'relative',
          }}
        >
          <Collapsible
            isOpen={isOpen}
            collapseHeight={40}
            style={{ ...maskStyle }}
            onInnerStateOpen={bool => {
              if (isOpen !== bool) {
                this.setState({
                  isOpen: bool,
                });
              }
            }}
          >
            {collapsed}
          </Collapsible>
          {isOpen ? null : (
            <a onClick={this.toggle} style={{ ...linkStyle }}>
              + Show More
            </a>
          )}
        </div>
      </>
    );
  }
}

export const CollapseHeight = () => <CusHeight />;

const Child = ({ onClick }) => {
  const [isCOpen, setIsCOpen] = useState(false);
  return (
    <div>
      <div
        style={{
          display: isCOpen ? 'block' : 'none',
          height: 200,
          background: 'green',
        }}
      >
        child
      </div>
      <Button
        onClick={() => {
          setIsCOpen(!isCOpen);
          onClick();
        }}
      >
        Toggle
      </Button>
    </div>
  );
}; // dynamic update content, children comp need to separate from parent to avoid rerender of entire comp

const DynamDemo = () => {
  const [isOpen, setIsOpen] = useState(true);
  const [reCalcKey, setReCalcKey] = useState(0);
  return (
    <div>
      <Button onClick={() => setIsOpen(!isOpen)}>折叠</Button>
      <Collapsible isOpen={isOpen}>
        <div>
          <div
            style={{
              height: 200,
              background: 'blue',
            }}
          >
            father
          </div>
          <Child onClick={() => setReCalcKey(reCalcKey + 1)} />
        </div>
      </Collapsible>
    </div>
  );
};

export const DynamicCollapsible = () => <DynamDemo />;