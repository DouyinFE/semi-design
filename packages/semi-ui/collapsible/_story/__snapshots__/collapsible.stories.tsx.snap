// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible 1`] = `
<Demo>
  <div>
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      prefixCls="semi-button"
      size="default"
      theme="light"
      type="primary"
    >
      <button
        aria-disabled={false}
        className="semi-button semi-button-primary semi-button-light"
        disabled={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="button"
      >
        <span
          className="semi-button-content"
          onClick={[Function]}
          x-semi-prop="children"
        >
          显示更多
        </span>
      </button>
    </Button>
    <Collapsible
      collapseHeight={0}
      duration={250}
      fade={false}
      isOpen={false}
      keepDOM={false}
      lazyRender={false}
      motion={true}
    >
      <div
        className="semi-collapsible-wrapper"
        onTransitionEnd={[Function]}
        style={
          Object {
            "height": 0,
            "opacity": 1,
            "overflow": "hidden",
            "transitionDuration": "0ms",
          }
        }
      >
        <div
          style={
            Object {
              "overflow": "hidden",
            }
          }
          x-semi-prop="children"
        />
      </div>
    </Collapsible>
  </div>
</Demo>
`;
